File: RRComSSys/Projects/CVM/RRComSSys.CVM/src/edu/fiu/cis/rrcomssys/cvm/mte/xcmlmodel/Medium.java
package edu.fiu.cis.rrcomssys.cvm.mte.xcmlmodel;

import org.simpleframework.xml.Attribute;

public class Medium
{
	/* Attributes */
	@Attribute(name = "mediumTypeName", required = true)
	private String _name = "";

	@Attribute(name = "suggestedApplication", required = false)
	private String _suggestedApplication = "";

	@Attribute(name = "voiceCommand", required = false)
	private String _voiceCommand = "";

	@Attribute(name = "derivedFromBuiltInType", required = true)
	private CapabilityType _capabilityType = CapabilityType.Text;

	/* Constructors & Initializers */
	public Medium()
	{}

	/* Properties */
	public String getName()
	{
		return _name;
	}

	public void setName(String value)
	{
		_name = value;
	}

	public String getSuggestedApplication()
	{
		return _suggestedApplication;
	}

	public void setSuggestedApplication(String value)
	{
		_suggestedApplication = value;
	}

	public String getVoiceCommand()
	{
		return _voiceCommand;
	}

	public void setVoiceCommand(String value)
	{
		_voiceCommand = value;
	}

	public CapabilityType getBuiltInType()
	{
		return _capabilityType;
	}

	public void setBuiltInType(CapabilityType value)
	{
		_capabilityType = value;
	}
}

File: trunk/RRComSSys/Projects/CVM/RRComSSys.CVM.ObjectModel/XCMLModel/Medium.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace RRComSSys.CVM.ObjectModel.XCMLModel
{
	public class Medium : IXCMLItem
	{
		#region Member Variables
		private String _name = "";
		private String _suggestedApplication = "";
		private String _voiceCommand = "";
		private CapabilityType _capabilityType = CapabilityType.Text;
		#endregion

		#region Properties
		[XmlAttribute(AttributeName = "mediumTypeName")]
		public String Name
		{
			get { return _name; }
			set { _name = value; }
		}

		[XmlAttribute(AttributeName = "suggestedApplication")]
		public String SuggestedApplication
		{
			get { return _suggestedApplication; }
			set { _suggestedApplication = value; }
		}

		[XmlAttribute(AttributeName = "voiceCommand")]
		public String VoiceCommand
		{
			get { return _voiceCommand; }
			set { _voiceCommand = value; }
		}

		[XmlAttribute(AttributeName = "derivedFromBuiltInType")]
		public CapabilityType BuiltInType
		{
			get { return _capabilityType; }
			set { _capabilityType = value; }
		}
		#endregion
	}
}
